node/src/actors/chain_manager/handlers.rs

```diff
@@ -834,6 +834,35 @@ impl PeersBeacons {
             .collect()
     }
 
+    /// Remove beacons that are old or has a different hash if is the same super epoch
+    pub fn remove_no_consensus_beacons(&mut self, superbeacon: CheckpointBeacon) {
+        let mut valid_peers = vec![];
+
+        for (addr, last_beacon) in &self.pb {
```

Comment:
I think this for loop can be rewritten to mutate the elements in place:

```suggestion
        for (addr, last_beacon) in &mut self.pb {
```

The only thing that is needed is to set the last_beacon to None in some cases, so this should work:

    *last_beacon = None;